test/strong_consistency_test.go

```diff
@@ -0,0 +1,365 @@
+						map[string]interface{}{
+							"name":               "sc1",
+							"memory-size":        1000955200,
+							"replication-factor": 5,
+							"strong-consistency": true,
+							"storage-engine": map[string]interface{}{
+								"type": "memory",
+							},
+						},
+					},
+				},
+			}
+
+			aeroCluster.Spec.RackConfig.Racks = racks
+
+			err := deployCluster(k8sClient, ctx, aeroCluster)
+			Expect(err).To(HaveOccurred())
+		})
+	})
+})
+
+// roster: node-id@rack-id
+
+func validateLifecycleOperationInSCCluster(
+	ctx goctx.Context, clusterNamespacedName types.NamespacedName, scNamespace string,
+) {
+	By("Scaleup")
+	err := scaleUpClusterTest(k8sClient, ctx, clusterNamespacedName, 3)
+	Expect(err).ToNot(HaveOccurred())
+
+	validateRoster(k8sClient, ctx, clusterNamespacedName, scNamespace)
+
+	By("Set roster blockList")
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	Expect(err).ToNot(HaveOccurred())
+	aeroCluster.Spec.RosterBlockList = []string{"1A0", "2A7"}
+	err = updateCluster(k8sClient, ctx, aeroCluster)
+	Expect(err).ToNot(HaveOccurred())
+
+	validateRoster(k8sClient, ctx, clusterNamespacedName, scNamespace)
+
+	By("ScaleDown")
+	err = scaleDownClusterTest(k8sClient, ctx, clusterNamespacedName, 2)
+	Expect(err).ToNot(HaveOccurred())
+
+	validateRoster(k8sClient, ctx, clusterNamespacedName, scNamespace)
+
+	By("RollingRestart")
+	err = rollingRestartClusterTest(logger, k8sClient, ctx, clusterNamespacedName)
+	Expect(err).ToNot(HaveOccurred())
+
+	validateRoster(k8sClient, ctx, clusterNamespacedName, scNamespace)
+
+	By("Upgrade/Downgrade")
+	// don't change image, it upgrades, check old version
+	err = upgradeClusterTest(
+		k8sClient, ctx, clusterNamespacedName, prevImage,
+	)
+	Expect(err).ToNot(HaveOccurred())
+
+	validateRoster(k8sClient, ctx, clusterNamespacedName, scNamespace)
+}
+
+func validateRoster(k8sClient client.Client, ctx goctx.Context, clusterNamespacedName types.NamespacedName, aeroNamespace string) {
+	aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+	Expect(err).ToNot(HaveOccurred())
+
+	hostConns, err := newAllHostConn(logger, aeroCluster, k8sClient)
+	Expect(err).ToNot(HaveOccurred())
+
+	rosterNodesMap, err := getRoster(hostConns[0], getClientPolicy(aeroCluster, k8sClient), aeroNamespace)
+	Expect(err).ToNot(HaveOccurred())
+
+	rosterStr := rosterNodesMap["roster"]
+	rosterList := strings.Split(rosterStr, ",")
+
+	// Check2 roster+blockList >= len(pods)
+	if len(rosterList)+len(aeroCluster.Spec.RosterBlockList) < len(aeroCluster.Status.Pods) {
+		err := fmt.Errorf("roster len not matching pods list. roster %v, blockList %v, pods %v", rosterList, aeroCluster.Spec.RosterBlockList, aeroCluster.Status.Pods)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	for _, rosterNode := range rosterList {
+		nodeID := strings.Split(rosterNode, "@")[0]
+
+		// Check1 roster should not have blocked pod
+		contains := v1beta1.ContainsString(aeroCluster.Spec.RosterBlockList, nodeID)
```

Comment:
This `aeroCluster.Spec.RosterBlockList` can be converted into a set for fast lookup here and in the below for loop